portable/GCC/ARM_CR5/port.c

```diff
@@ -255,12 +281,31 @@ StackType_t * pxPortInitialiseStack( StackType_t * pxTopOfStack,
     /* The task will start with a critical nesting count of 0 as interrupts are
      * enabled. */
     *pxTopOfStack = portNO_CRITICAL_NESTING;
-    pxTopOfStack--;
 
-    /* The task will start without a floating point context.  A task that uses
-     * the floating point hardware must call vPortTaskUsesFPU() before executing
-     * any floating point instructions. */
-    *pxTopOfStack = portNO_FLOATING_POINT_CONTEXT;
+    #if( configUSE_TASK_FPU_SUPPORT == 1 )

```

Comment:
It is possible that this port is being compiled for a target without an FPU, in which case configUSE_TASK_FPU_SUPPORT may not be defined. This is probably an issue in the original Cortex-A port as well. This code as-is does not compile when configUSE_TASK_FPU_SUPPORT is left undefined or set to 0.

It might be preferable to use the configENABLE_FPU option from the ARMv8M ports and make FPU management on by default for each task rather than opt-in.

@RichardBarry and @aggarg Do you have a preference here?

Reply:
Oh, I see what you, @paulbartell , mean. I did take a look on the `configENABLE_FPU` option, and I strongly agree with you. Let me know, if you want the `configENABLE_FPU`-option implemented on both ports (Cortex-R5 and A9), I would gladly do it.

Reply:
I think we should merge this PR as is and then create a separate PR which should allow setting `configUSE_TASK_FPU_SUPPORT` to 0. The `configUSE_TASK_FPU_SUPPORT` would then have the following meaning - 
* `0` - No FPU support.
*  `1` - Tasks individually register their intent to have a FPU context.
* `2` - All tasks have FPU context.